tests/Unit/Domain/Test_Domain.cpp

```diff
@@ -137,3 +139,133 @@ void test_1d_domains() {
 }  // namespace
 
 SPECTRE_TEST_CASE("Unit.Domain.Domain", "[Domain][Unit]") { test_1d_domains(); }
+SPECTRE_TEST_CASE("Unit.Domain.Domain.Rectilinear1D1", "[Domain][Unit]") {
```

Comment:
Is there any reason not to have two functions that are called by `Rectilinear1D`? Fewer tests to run means tests complete faster :)